spec/controllers/queue_items_controller_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'spec_helper'
+
+describe QueueItemsController do 
+  describe 'GET index' do 
+    it 'sets @queue_items to the queue_items of the logged in user' do 
+      alice = Fabricate(:user)
+      session[:user_id] = alice.id
+      video1 = Fabricate(:video)
+      video2 = Fabricate(:video)
+      queue_item1 = Fabricate(:queue_item, user: alice, video: video1)
+      queue_item2 = Fabricate(:queue_item, user: alice, video: video2)
+      get :index
+      expect(assigns(:queue_items)).to match_array([queue_item1, queue_item2])
+    end
+    
+    it 'redirects to the sign in page for unauthenticated users' do
+      get :index
+      expect(response).to redirect_to signin_path
+    end
+  end
+  
+  describe 'POST create' do 
+    let(:user1){ Fabricate(:user) }
+    let(:video){ Fabricate(:video) }
+    
+    it 'redirects to the my queue page' do 
+      session[:user_id] = user1.id 
+      post :create, video_id: Fabricate(:video).id, user_id: user1.id
+      expect(response).to redirect_to my_queue_path   
+    end
+    
+    it 'creates a queue item' do
+      session[:user_id] = user1.id 
+      post :create, video_id: Fabricate(:video).id, user_id: user1.id
```

Comment:
Passing in `user_id` should not be needed. Actually you should even delete it, because this looks like a security weak point. A user can open browser and change the id of user assigning the QueueItem to someone totally different. Your controller is _aware_ of current_user who is putting the item to the queue.